Osiris/Hacks/Misc.cpp

```diff
@@ -335,6 +345,111 @@ void Misc::spectatorList() noexcept
     ImGui::End();
 }
 
+struct DamageData {
+    DamageData(int damage, int playerHandle) : damage{ damage }, playerHandle{ playerHandle } {}
+    
+    int damage;
+    int playerHandle;
+};
+
+static std::vector<DamageData> damageCount;
+
+void Misc::damageList() noexcept
+{
+    static int rows_count;
+
+    if (!miscConfig.damageList.enabled)
+        return;
+
+    if ((damageCount.size() == 0 || !interfaces->engine->isInGame() || !interfaces->engine->isConnected()) && !gui->isOpen())
+        return;
+    
+    if (miscConfig.damageList.pos != ImVec2{}) {
+        ImGui::SetNextWindowPos(miscConfig.damageList.pos);
+        miscConfig.damageList.pos = {};
+    }
+
+    if (miscConfig.damageList.size != ImVec2{}) {
+        ImGui::SetNextWindowSize(ImClamp(miscConfig.damageList.size, {}, ImGui::GetIO().DisplaySize));
+        miscConfig.damageList.size = {};
+    }
+
+    ImGuiWindowFlags windowFlags = ImGuiWindowFlags_NoCollapse;
+    if (!gui->isOpen())
+        windowFlags |= ImGuiWindowFlags_NoInputs;
+    if (miscConfig.damageList.noTitleBar)
+        windowFlags |= ImGuiWindowFlags_NoTitleBar;
+
+    if (!gui->isOpen())
+        ImGui::PushStyleColor(ImGuiCol_TitleBg, ImGui::GetColorU32(ImGuiCol_TitleBgActive));
+
+    ImGui::PushStyleVar(ImGuiStyleVar_WindowTitleAlign, { 0.5f, 0.5f });
+    ImGui::Begin("Damage list", nullptr, windowFlags);
+    ImGui::PopStyleVar();
+
+    if (!gui->isOpen())
+        ImGui::PopStyleColor();
+
+    rows_count = 0;
+
+    for (size_t i = 0; i < damageCount.size(); i++) {
+
+        if (rows_count >= miscConfig.damageList.maxRows)
+            break;
+
+        if (damageCount[i].damage == 0)
+            continue;
+
+        if (const auto playerdata = GameData::playerByHandle(damageCount[i].playerHandle)) {
```

Comment:
You're missing a `GameData::Lock`.

Reply:
Thanks.
Btw, is it possible to get PlayerData for the local player without this?
https://github.com/danielkrupinski/Osiris/blob/ff6519e0a335517f2b5031588a05f39453940aa9/Osiris/GameData.cpp#L113-L114
I tried this:
That's ok?
https://github.com/danielkrupinski/Osiris/blob/ff6519e0a335517f2b5031588a05f39453940aa9/Osiris/GameData.h#L112

Reply:
We have `LocalPlayerData` for that.

Reply:
`LocalPlayerData` does not contain `getAvatarTexture()`